### ISSUE TITLE
Related getter has an undefined item after deleting the related resource

### ISSUE THREAD

**Issue by steven-fox** [reststate/reststate-vuex#180]
### Setup

We have a simple situation where a resource has many related resource records. Think a blog `post` and its `comments`.

We have a Vue component to handle the `comments`. We pass the `post` object into the component as a prop called `parent` for relationship identifier purposes. From there, we have the following snippet:
```javascript
// Comments.vue
<template>
...
  <div v-for="comment in comments" :key="comment.id">
      <comment :comment="comment" v-on:delete-comment="deleteComment" />
  </div>
...
</template>

<script>
...
computed: {
  comments() {
    return this.$store.getters['comments/related']({ parent: this.parent })
  }
},
methods: {
  loadComments() {
    this.$store.dispatch('comments/loadRelated', {
      parent: this.parent
    })
  },
  deleteComment(comment) {
    this.$store.dispatch('comments/delete', comment)
}
...
```

Pretty standard setup. We're using the blog post as the `parent` and retrieving the comments related to it. From there, we have a computed property called `comments` from the store's getter and that is used to render a `v-for` of `comment` components. The `comment` component will emit a event when the user clicks to delete their comment. To do so, we're simply dispatching the store's action.

**Important notice:** this bug appears specifically when using the `related` getter.

### The Bug
Having a collection of `comments` and then deleting a single item from that collection will trigger the bug. When this happens, we get an error in the console saying `comment is undefined in Comments.vue`. Sure enough, there is still an item in the computed `comments` but it's value is `undefined`. Ie, the index key in the array exists but it's value is undefined. Here's a screenshot:

<img width="252" alt="image" src="https://user-images.githubusercontent.com/62109327/78939951-0ae05580-7a83-11ea-8b15-a3b20e043da1.png">

What's interesting is that the `state` object itself has the following results after performing the comment's delete:

<img width="499" alt="image" src="https://user-images.githubusercontent.com/62109327/78940184-6f9bb000-7a83-11ea-8b14-0be34b4002b0.png">

As you can see, `comments.records` is accurately in a state of an empty array (we started with 1 comment for this example). Meanwhile, `comments.related` is still identifying the `relatedId` of the `comment` before it was deleted.

### Hypothesized Source of the Problem
I'm not a Vue or javascript expert - far from it in fact. However, I believe the problem stems from something like this:

1. When the `REMOVE_RECORD` action/mutation is called, the record is removed from the state's `records` array.
2. From there, the `related` getter then attempts to 
```javascript
// getters.related...
const related = state.related.find(matches(relationshipIndex)); // line 371
...
return ids.map(id => state.records.find(record => record.id === id)); // line 377
...
```
3. However, because the record doesn't actually exist, it yields an array with a key assigned but an undefined value.
4. The computed property reacts and is thus undefined for the `v-for` loop.

### Conclusion
I _believe_ this is a bug with the package and not an implementation mistake. However, being the non-expert I am, that may very well be the case. Nonetheless, I'd greatly appreciate a response from anyone looking, or @CodingItWrong, to see if I may be on to something here. Unfortunately, I'm not versed enough in Vue reactivity nor Javascript to assist with a worthy PR attempt. Perhaps there are 2 potential solutions:

1. Update the `REMOVE_RECORD` mutation to find locations where that resource is stored within another resource's `related` records and remove from those arrays. Then, when the `related` getter is performed again, it won't try to map that object that no longer exists (line 371).
2. Update the `related` getter so that we first ensure a `state.records` can be found before/during the `map` operation (line 377 and 380).

**Comment by steven-fox** [comment]
### How we're getting around this issue

In case it matters to anyone, we have gotten around this issue by using a data + method approach in our component. Instead of using the getter on a computed property, we initialize the component with an empty comments array and then manually clone the store's getter results into that data variable each time an event occurs. It's basically like manually performing the reactivity needed. When adding/removing elements from a Vuex array, this approach has been explained on the interwebs. This isn't a _clean_ approach, perhaps, but it does seem to produce the desired outcome without any undefined variables sneaking in.

```javascript
// Comments.vue
...
<script>
data() {
  return {
    comments: []
  }
},
methods: {
  loadComments() {
    this.$store
      .dispatch('comments/loadRelated', { parent, options })
      .then(() => {
         this.comments = this.$store.getters['comments/related']({ parent })
    })
  }
},
mounted() {
  this.loadComments()
}
...

**Comment by CodingItWrong** [comment]
@steven-fox Thanks for reporting this. This has been fixed in #185 and released in `0.2.1`. Please give it a try and let me know if the fix works for you!

**Comment by steven-fox** [comment]
@CodingItWrong Thanks Josh. We'll pull down the newest package version and give it a whirl. I will report here if anything seems broken, but I'm not anticipating that to be the case.

Thank you for the support and update. :clap:
